x-pack/plugins/stack_alerts/public/alert_types/es_query/search_source_threshold_expression.tsx

```diff
@@ -0,0 +1,198 @@
+ * 2.0.
+ */
+
+import React, { Fragment, useCallback, useEffect, useState } from 'react';
+import './search_source_threshold_expression.scss';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiSpacer, EuiTitle, EuiExpression, EuiPopover, EuiText } from '@elastic/eui';
+import { Filter, ISearchSource } from '../../../../../../src/plugins/data/common';
+import { QueryStringInput } from '../../../../../../src/plugins/data/public';
+import { EsQueryAlertParams } from './types';
+import {
+  ForLastExpression,
+  RuleTypeParamsExpressionProps,
+  ThresholdExpression,
+} from '../../../../triggers_actions_ui/public';
+import { DEFAULT_VALUES } from './constants';
+import { ReadOnlyFilterItems } from '../components/read_only_filter_items';
+
+export const SearchSourceThresholdExpression = ({
+  ruleParams,
+  setRuleParams,
+  setRuleProperty,
+  data,
+  errors,
+}: RuleTypeParamsExpressionProps<EsQueryAlertParams>) => {
+  const { thresholdComparator, threshold, timeWindowSize, timeWindowUnit } = ruleParams;
+
+  const getDefaultParams = () => {
+    const defaults = {
+      timeWindowSize: timeWindowSize ?? DEFAULT_VALUES.TIME_WINDOW_SIZE,
+      timeWindowUnit: timeWindowUnit ?? DEFAULT_VALUES.TIME_WINDOW_UNIT,
+      threshold: threshold ?? DEFAULT_VALUES.THRESHOLD,
+      thresholdComparator: thresholdComparator ?? DEFAULT_VALUES.THRESHOLD_COMPARATOR,
+    };
+
+    return { ...ruleParams, ...defaults, searchType: 'searchSource' };
+  };
+
+  const [currentAlertParams, setCurrentAlertParams] = useState<EsQueryAlertParams>(
+    getDefaultParams()
+  );
+
+  const setParam = useCallback(
+    (paramField: string, paramValue: unknown) => {
+      setCurrentAlertParams((currentParams) => ({
+        ...currentParams,
+        [paramField]: paramValue,
+      }));
+      setRuleParams(paramField, paramValue);
+    },
+    [setRuleParams]
+  );
+
+  useEffect(() => {
+    setRuleProperty('params', currentAlertParams);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const [usedSearchSource, setUsedSearchSource] = useState<ISearchSource | undefined>();
+
+  const { searchConfiguration } = ruleParams;
+
+  // Note that this PR contains a limited way to edit query and filter
+  // But it's out of scope for the MVP
+  const [showQueryBar, setShowQueryBar] = useState<boolean>(false);
+  const [showFilter, setShowFilter] = useState<boolean>(false);
+
+  useEffect(() => {
+    async function initSearchSource() {
+      const loadedSearchSource = await data.search.searchSource.create(searchConfiguration);
+      setUsedSearchSource(loadedSearchSource);
+    }
+    if (searchConfiguration) {
+      initSearchSource();
+    }
+  }, [data.search.searchSource, searchConfiguration]);
+
+  if (!usedSearchSource) {
+    // there should be a loading indicator
+    return null;
+  }
+
+  const filters = (usedSearchSource.getField('filter') as Filter[]).filter(
+    ({ meta }) => !meta.disabled
```

Comment:
There is no need of disabled filters to be displayed in the alert flyout, since result set will not be affected by that filters.